tests/test_contracts/test_contracts.py

```diff
@@ -853,3 +853,245 @@ def return_bool() -> bool:
+
+    process_float_key_dict({1.0: 10, 2.5: 20})
+
+
+def test_parameter_union_int_float_pass() -> None:
+    """Testing when the function expects Union[int, float] and it receives an int."""
+
+    @check_contracts
+    def accept_union(value: Union[int, float]) -> None:
+        pass
+
+    accept_union(10)
+
+
+def test_union_with_matching_collection_types() -> None:
+    """Test cases where the provided collection type matches one of the types specified in a union
+    (e.g., expecting dict[int, str] | list[int], and given list[int])."""
+
+    @check_contracts
+    def process_dict(value: Union[dict[int, str], list[int]]) -> str:
+        return str(value)
+
+    @check_contracts
+    def process_tuple(value: Union[tuple[int, str], list[int]]) -> str:
+        return f"{value[0]}, {value[1]}"
+
+    @check_contracts
+    def process_set(value: Union[set[int], list[int]]) -> str:
+        return str(value)
+
+    process_dict([1, 2, 3])
+    process_tuple([1, 2, 3])
+    process_set([1, 2, 3])
+
+
+def test_parameter_list_float_int_without_strict(disable_strict_numeric_types) -> None:
+    """
+    Test that an AssertionError is not raised when a function expects list[float] but receives list[int],
+    while STRICT_NUMERIC_TYPES is disabled.
+    """
+
+    @check_contracts
+    def process_float_list(data: list[float]) -> float:
+        return sum(data)
+
+    process_float_list([1, 2, 3])
+
+
+def test_parameter_dict_float_int_key_without_strict(disable_strict_numeric_types) -> None:
+    """
+    Test that an AssertionError is not raised when a function expects dict[float, int] but receives dict[int, int],
+    while STRICT_NUMERIC_TYPES is disabled.
+    """
+
+    @check_contracts
+    def process_float_key_dict(data: dict[float, int]) -> int:
+        return sum(data.values())
+
+    process_float_key_dict({1: 10, 2: 20})
+
+
+def test_parameter_float_complex() -> None:
+    """
+    Testing when the function expects a float but receives a complex number.
+    This should raise an AssertionError with strict type checking enabled.
+    """
+
+    @check_contracts
+    def process_float(value: float) -> float:
+        return value.real
+
+    with pytest.raises(AssertionError):
+        process_float(3 + 4j)
+
+
+def test_parameter_float_complex_without_strict(disable_strict_numeric_types) -> None:
+    """
+    Test that an AssertionError is not raised when a function expects a float but receives a complex number,
+    while STRICT_NUMERIC_TYPES is disabled.
+    """
+
+    @check_contracts
+    def process_float(value: complex) -> float:
+        return value.real
+
+    process_float(3.0)
+
+
+def test_parameter_tuple_fixed_length_types() -> None:
+    """Testing when the function expects a fixed-length tuple with specific types (tuple[int, str])."""
+
+    @check_contracts
+    def accept_fixed_tuple(value: tuple[int, str]) -> str:
+        return f"{value[0]}: {value[1]}"
+
+    accept_fixed_tuple((5, "test"))
```

Comment:
separate these into separate test cases (like you did all other test cases; tuples aren't special)